packages/batch-submitter/package.json

```diff
@@ -13,10 +13,10 @@
     "build": "tsc -p ./tsconfig.build.json",
     "clean": "rimraf cache/ dist/ ./tsconfig.build.tsbuildinfo",
     "lint": "yarn lint:fix && yarn lint:check",
-    "lint:fix": "prettier --config .prettierrc.json --write \"hardhat.config.ts\" \"{src,exec,test}/**/*.ts\"",
-    "lint:check": "eslint -c .eslintrc.js --ext .ts --format stylish .",
-    "test": "hardhat test --show-stack-traces",
-    "pre-commit": "lint-staged"
+    "pre-commit": "lint-staged",
```

Comment:
This pre-commit hook here isn't producing the same results as running `lint:check`. I am testing this by messing with the `optimism/packages/contracts/test/contracts/OVM/bridge/base/OVM_L1CrossDomainMessenger.spec.ts` test file.

Reply:
To test this I made the following diff:

```diff
i/packages/contracts/test/contracts/OVM/bridge/base/OVM_L1CrossDomainMessenger.spec.ts
+++ w/packages/contracts/test/contracts/OVM/bridge/base/OVM_L1CrossDomainMessenger.spec.ts
@@ -23,11 +23,10 @@ import {
 import { keccak256 } from 'ethers/lib/utils'
 import { predeploys } from '../../../../../src'
 
-const MAX_GAS_LIMIT = 8_000_000
+let MAX_GAS_LIMIT = 8_000_000
 
 const deployProxyXDomainMessenger = async (
-  addressManager: Contract,
-  l1XDomainMessenger: Contract
+  addressManager: Contract,l1XDomainMessenger: Contract
 ): Promise<Contract> => {
   await addressManager.setAddress(
     'OVM_L1CrossDomainMessenger',
```

Then ran `yarn pre-commit` inside the `contracts` package. The output is:

```
/Users/liam/Projects/optimism/optimism/packages/contracts/test/contracts/OVM/bridge/base/OVM_L1CrossDomainMessenger.spec.ts
  26:5   error    'MAX_GAS_LIMIT' is never reassigned. Use 'const' instead  prefer-const
  29:28  warning  Insert `⏎··`   
```

Then I ran `yarn lint:check` and I see

```
/Users/liam/Projects/optimism/optimism/packages/contracts/test/contracts/OVM/bridge/base/OVM_L1CrossDomainMessenger.spec.ts
  26:5   error    'MAX_GAS_LIMIT' is never reassigned. Use 'const' instead  prefer-const
  29:28  warning  Insert `⏎··`                                              prettier/prettier
```

Which seems to match — do you have an example of something not matching?

Reply:
This is already merged before I had a chance to test again but we'll find out soon I guess